website/static/js/registrationUtils.js

```diff
@@ -396,8 +396,7 @@ Draft.prototype.register = function(data) {
         .done(function(response) {
             if (response.status === 'initiated') {
                 window.location.assign(response.urls.registrations);
-            }
-            else if (response.status === 'error') {
+            } else if (response.status === 'error') {
```

Comment:
You should be able to remove this error handling logic from the `.done` branch, as the promise should make sure it's never reached. 

jqXHR (which are promises) will be rejected for any HTTP error responses (400 and 500 responses). (h/t Steve) This means that a bad error code would fir ethe `fail` callbacks instead.

self.urls.register seems to map to [this view](https://github.com/HarryRybacki/osf.io/blob/feature/prereg-prize/website/project/views/drafts.py#L63), which only produces `'status': 'error'` alongside an appropriate error code.